packages/core-data/src/selectors.ts

```diff
@@ -69,41 +131,44 @@ export function getAuthors( state, query ) {
 /**
  * Returns the current user.
  *
- * @param {Object} state Data state.
+ * @param  state Data state.
  *
- * @return {Object} Current user object.
+ * @return Current user object.
  */
-export function getCurrentUser( state ) {
+export function getCurrentUser( state: State ): User< 'edit' > {
```

Comment:
The context here should be `'view'` instead of `'edit'`, otherwise it will set the wrong types/properties which are not available in current user object, for example, `user.capabilities`.

Reply:
Created #68045 to fix it.